client/pages/home.tsx

```diff
@@ -80,12 +46,19 @@ const GET_GOALS_QUERY = gql`
 `;
 
 const HomePage = () => {
-  // TODO: Replace with correct logic here
-  const { data, error } = useQuery(GET_GOALS_QUERY);
-  // To peek inside the output of this query,
-  // simply console log both data and error
+  const { data, error, loading } = useQuery(GET_GOALS_QUERY);
+
+  if (
+    error &&
+    error.message === 'not authenticated' &&
+    typeof window !== 'undefined'
+  ) {
+    window.location.pathname = '/login';
```

Comment:
Should use next's router.replace() here so the current path gets replaced with the new path rather than pushing them to a new path. Also because the login page isn't using SSR